### ISSUE TITLE
Approximate Counts

### ISSUE THREAD

**Issue by JLockerman** [timescale/timescaledb-toolkit#6]
## What's the functionality you would like to add ##
Approximate counts for the various values in a dataset, such as Count-Min Sketch.

## How would the function be used ##
```SQL
SELECT approx_count(value, 'ERROR') FROM data;
approx_count
-----------------
              42

CREATE MATERIALIZED VIEW distinct_per_hour
WITH (timescaledb.continuous) AS
SELECT time_bucket(INTERVAL '1 hour', time) AS bucket,
    approx_counts(value),
FROM data
GROUP BY bucket;

SELECT approx_count(approx_counts, 'IDLE') FROM distinct_per_hour LIMIT 4;
         bucket         | approx_count
------------------------+------------------
   2020-01-03 01:00:00  |               10
   2020-01-03 02:00:00  |                8
   2020-01-03 03:00:00  |                8
   2020-01-03 04:00:00  |               11

```

## Why should this feature be added?  ##
Mainly for continuous aggregates and other circumstances where you want to chose which elements to analyze after aggregation.

*Note: edited by David as the numeric frequencies seemed somewhat confusing in the example

**Comment by davidkohn88** [comment]
Discussion in #32.

**Comment by rtwalker** [comment]
Closed by #458
